frame/election-provider-multi-phase/src/signed.rs

```diff
@@ -349,6 +351,31 @@ impl<T: Config> SignedSubmissions<T> {
 	}
 }
 
+/// Type that can be used to calculate the deposit base for signed submissions.
+///
+/// The deposit base is calculated as a geometric progression based on the number of signed
+/// submissions in the queue. The size of the queue represents the progression term.
+pub struct GeometricDepositBase<T> {
+	_marker: PhantomData<T>,
+}
+
+impl<T: Config> Convert<usize, BalanceOf<T>> for GeometricDepositBase<T> {
+	// Calculates the base deposit as a geometric progression based on the number of signed
+	// su Calculates.
+	//
+	// The nth term is obtained by calculating `base * (1 + increase_factor)^nth`. Example: factor
+	// 5, with initial deposit of 1000 and 10% of increase factor is 1000 * (1 + 0.1)^5.
+	fn convert(queue_len: usize) -> BalanceOf<T> {
+		let increase_factor: FixedU128 =
+			<Percent as Into<FixedU128>>::into(T::SignedDepositBaseIncreaseFactor::get()) +
+				sp_runtime::traits::One::one();
```

Comment:
This is a little more concise way to do it.
```suggestion
			FixedU128::from_u32(1) + T::SignedDepositBaseIncreaseFactor::get().into();
```